pygeos/decorators.py

```diff
@@ -16,19 +16,23 @@ def __init__(self, version):
         self.version = tuple(int(x) for x in version.split("."))
 
     def __call__(self, func):
+        msg = "This function requires at least GEOS {}.{}.{}.".format(*self.version)
         if lib.geos_version < self.version:
-            msg = "'{}' requires at least GEOS {}.{}.{}".format(
-                func.__name__, *self.version
-            )
 
             @wraps(func)
             def wrapped(*args, **kwargs):
                 raise UnsupportedGEOSOperation(msg)
 
-            wrapped.__doc__ = msg
-            return wrapped
         else:
-            return func
+            wrapped = func
+
+        # Add a note about the version constraint after the first line of
+        # the docstring.
+        if wrapped.__doc__:
```

Comment:
It seems like this would be more useful where `lib.geos_version < self.version`; if the current install of pygeos / GEOS already satisfies this requirement, this is not necessary info unless needed for the docs on readthedocs.